oracle_cardano/processor/tx_processors/success/bridging_requested_processor_skyline.go

```diff
@@ -0,0 +1,355 @@
+		}
+
+		nativeCurrencyAmountSum.Add(nativeCurrencyAmountSum, new(big.Int).SetUint64(totalNativeCurrencyAmountToBridge))
+		wrappedTokenAmountSum.Add(wrappedTokenAmountSum, new(big.Int).SetUint64(totalWrappedTokenAmountToBridge))
+	}
+
+	if foundAUtxoValueBelowMinimumValue {
+		return fmt.Errorf("found a utxo value below minimum value in metadata receivers: %v", metadata)
+	}
+
+	if foundAnInvalidReceiverAddr {
+		return fmt.Errorf("found an invalid receiver addr in metadata: %v", metadata)
+	}
+
+	feeExchangeDiff := float64(metadata.FeeAmount.DestAmount)*exchangeRate - float64(metadata.FeeAmount.SrcAmount)
+	if (feeExchangeDiff > 0 && feeExchangeDiff > exchangeMargin) ||
+		(feeExchangeDiff < 0 && feeExchangeDiff < -exchangeMargin) {
+		return fmt.Errorf("found an exchage rate error if fee metadata: %v", metadata)
+	}
+
+	// update fee amount if needed with sum of fee address receivers
+	feeAmount := metadata.FeeAmount.SrcAmount + feeSum
+	nativeCurrencyAmountSum.Add(nativeCurrencyAmountSum, new(big.Int).SetUint64(feeAmount))
+
+	if feeAmount < cardanoSrcConfig.MinFeeForBridging {
+		return fmt.Errorf("bridging fee in metadata receivers is less than minimum: fee %d, minFee %d, metadata %v",
+			feeAmount, cardanoSrcConfig.MinFeeForBridging, metadata)
+	}
+
+	if nativeCurrencyAmountSum.Cmp(new(big.Int).SetUint64(multisigUtxo.Amount)) != 0 {
+		return fmt.Errorf("multisig amount is not equal to sum of receiver amounts + fee: expected %v but got %v",
+			multisigUtxo.Amount, nativeCurrencyAmountSum)
+	}
+
+	multisigWrappedTokenAmount := GetNativeTokenAmount(cardanoSrcConfig, multisigUtxo, metadata.DestinationChainID)
+
+	if wrappedTokenAmountSum.Cmp(new(big.Int).SetUint64(multisigWrappedTokenAmount)) != 0 {
+		return fmt.Errorf("multisig wrapped token is not equal to receiver wrapped token amount: expected %v but got %v",
+			multisigWrappedTokenAmount, wrappedTokenAmountSum)
+	}
+
+	if appConfig.BridgingSettings.MaxAmountAllowedToBridge != nil &&
+		appConfig.BridgingSettings.MaxAmountAllowedToBridge.Sign() > 0 &&
+		nativeCurrencyAmountSum.Cmp(appConfig.BridgingSettings.MaxAmountAllowedToBridge) == 1 {
+		return fmt.Errorf("sum of receiver amounts + fee: %v greater than maximum allowed: %v",
+			nativeCurrencyAmountSum, appConfig.BridgingSettings.MaxAmountAllowedToBridge)
+	}
+
+	return nil
+}
+
+func GetExchangeRate(sourceChainID string, destinationChainID string) (float64, error) {
+	if destinationChainID == "prime" {
+		return 0.5, nil
+	}
+
+	return 2, nil
+}
+
+func GetNativeTokenAmount(config *cCore.CardanoChainConfig, utxo *indexer.TxOutput, chainID string) uint64 {
+	amount := uint64(0)
+
+	var tokenName string
+
+	tokenNames := config.Destinations
+
+	for _, token := range tokenNames {
+		if token.Chain == chainID {
+			tokenName = token.SrcTokenName
+
+			break
+		}
+	}
+
+	for _, token := range utxo.Tokens {
+		fullTokenName := fmt.Sprintf("%s.%s", token.PolicyID, token.Name)
```

Comment:
line not needed

Reply:
do no use this like that, use utxo. TokenName()